pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -0,0 +1,157 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package portcache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/rules"
+)
+
+type NodePortData struct {
+	NodePort int
+	PodPort  int
+	PodIP    string
+	Status   int
+}
+
+type PortTable struct {
+	Table        map[int]NodePortData
+	StartPort    int
+	EndPort      int
+	PodPortRules rules.PodPortRules
+	tableLock    sync.RWMutex
+}
+
+func NewPortTable(start, end int) (*PortTable, bool) {
```

Comment:
idea for a future PR: we can do the same thing as kube-proxy for NodePorts and actually "hold" the port by binding a socket to it and keeping it alive

see https://github.com/kubernetes/kubernetes/blob/86f8c3ee91b6faec437f97e3991107747d7fc5e8/pkg/proxy/iptables/proxier.go#L1664

I know you had something like this originally, but IIRC you were just trying to bind to the port to check if it was available at that point in time, and not actually reserving the port indefinitely

Reply:
Sure Antonin. I'll take care of this in next PR.